### PR TITLE
review list beneficiaries cache

### LINKED ISSUE #529: clean endpoints cache when a beneficiary is added or removed
- use chainSubscriber to listen to the events
- check if this event came from a iM contract
- clean cache by redis key

### PR DESCRIPTION
**Pull request by dev-jotape**
This PR fixes #529 

## Changes
Clean cache when a beneficiary is added or removed

### CODE REVIEW

**Review by obernardovieira — CHANGES_REQUESTED** [review]
Just some minor cleanups 😉

File: packages/core/src/subscriber/chainSubscribers.ts [hunk 1]
```diff
@@ -11,21 +11,41 @@ class ChainSubscribers {
...
+    ) {
         this.provider = webSocketProvider;
         this.jsonRpcProvider = jsonRpcProvider;
         this.ifaceCommunityAdmin = new ethers.utils.Interface(
             contracts.CommunityAdminABI
         );
-
+        this.ifaceOldCommunity = new ethers.utils.Interface(
+            contracts.OldCommunityABI
+        );
```
[the remark this answers is not in the document]
Comment (dev-jotape) [reply packages/core/src/subscriber/chainSubscribers.ts]: ok, removed.

File: packages/core/src/subscriber/chainSubscribers.ts [hunk 2]
```diff
@@ -11,21 +11,41 @@ class ChainSubscribers {
...
         this.filterTopics = [
             [
                 ethers.utils.id(
                     'CommunityAdded(address,address[],uint256,uint256,uint256,uint256,uint256,uint256,uint256)'
                 ),
                 ethers.utils.id('CommunityRemoved(address)'),
+                ethers.utils.id('BeneficiaryAdded(address)'),
+                ethers.utils.id('BeneficiaryAdded(address,address)'),
+                ethers.utils.id('BeneficiaryRemoved(address)'),
+                ethers.utils.id('BeneficiaryRemoved(address,address)'),
```
[the remark this answers is not in the document]
Comment (dev-jotape) [reply packages/core/src/subscriber/chainSubscribers.ts]: ok, removed

**Review by obernardovieira — APPROVED** [review]
🥳

**Pull request merged**